### DISCUSSION TITLE
Access checked rows in details editor from master dialog and save

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#7303] [Help] [answered]
Hi All,
Please I have a scenario as below where I need to save the selected rows from two details grid when the save button on the master dialog is clicked.
How do I get access to the selected rows in the two grids to save.
Please any help will be appreciated

![image](https://github.com/user-attachments/assets/3baaf5f0-1148-478b-b019-47068589601f)

**Comment by rhemasystems** [comment]
Please any help will be greatly appreciated, I know its wrong to tag a user but am stuck at this point.
Please help @volkanceylan @VictorTomaili

**Comment by rhemasystems** [comment]
@minhhungit , please can you assist me

**Reply by minhhungit** [reply, marked as answer]
@rhemasystems  To achieve this, I guess you can get selected rows from this property `this.from.YourDetailGridField` (I don't remember exact the sub-method you can use). Next, you can customize the save entity by overriding the getSaveEntity() method: 

https://github.com/serenity-is/Serenity/blob/8e4d90fb072ceca3558c15d40a2a9c9c8e87746f/packages/corelib/src/ui/dialogs/entitydialog.tsx#L539

Once all the necessary information is prepared and sent to the server-side, you can implement your desired logic in the Save Handler to handle the data as needed.

**Reply by rhemasystems** [reply]
Thank you so much, below is what I did, I hope its the best way.

I created a public number variable as an array, and assign the selected ids to it in the constructor.

DetailEditor.tsx
```
public itemkeys: number[];
     .........
     .........

var keys = this.rowSelection.getSelectedKeys();
var items = keys.map(x => this.view.getItemById(toId(x)).Id).filter(x => x != null)
this.itemkeys = items; // assigned to public variable
```
Master dialog.tsx
`````
protected getSaveEntity() {
    var items = this.form.ItemList.itemkeys;
    ......
}
`````
